### ISSUE TITLE
Upload multiple reports with different flags

### ISSUE THREAD

**Issue by Ekleog-NEAR** [codecov/codecov-action#1176]
Hello!

I’ll start with, this is a duplicate of closed-by-author #243, and of the discussion that happened after #50 was closed.

I’d like to upload multiple reports with different flags. It currently seems to basically require calling the codecov actions multiple times. However, it results in a lot of copy-paste, and workarounds like using `steps: fromJson()` with a custom generator script is even more painful.

Would it be possible to upload multiple reports with different flags? I’m thinking of a syntax like:
```yml
- uses: actions/codecov-action@v??
  with:
    token: ${{ secrets.CODECOV_TOKEN }}
    fail_ci_if_error: true
    files:
      - { path: "unit-tests.json", flags: "unit-tests" }
      - { path: "integration-tests.json", flags: "integration-tests" }
```

What do you think?

**Comment by thomasrockhu-codecov** [comment]
@Ekleog-NEAR, no this is not possible as specified. You can instead run the action twice, each specifying a file and a flag

**Comment by Ekleog-NEAR** [comment]
@thomasrockhu-codecov Sorry for being unclear in my message!

I totally understand that it is currently not possible. What I’m wondering is, would you be interested in changing codecov-action, so that it becomes possible? It feels like a change that would be welcome in v4, and probably not be too hard to implement.

Right now our solution is indeed to [have multiple calls to the action](https://github.com/near/nearcore/pull/10250/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR297-R377), but it’s quite a lot of copy-paste, which is never good :/
